gluten-flink/runtime/src/main/java/org/apache/flink/client/StreamGraphTranslator.java

```diff
@@ -96,71 +104,121 @@ private JobGraph mergeGlutenOperators(JobGraph jobGraph) {
     return jobGraph;
   }
 
-  // A JobVertex may contain several operators chained like this: Source-->Op1-->Op2-->Sink.
+  // A JobVertex may contain several operators chained like this: Source-->Op1-->Op2-->Sink1.
+  //                                                                                -->Sink2.
   // If the operators connected all support translated to gluten, we merge them into
   // a single GlutenOperator to avoid data transferred between flink and native.
-  // Now we only support that one operator followed by at most one other operator.
+  // One operator may be followed by several other operators.
   private void buildGlutenChains(StreamConfig vertexConfig) {
     Map<Integer, StreamConfig> serializedTasks =
         vertexConfig.getTransitiveChainedTaskConfigs(userClassloader);
     Map<Integer, StreamConfig> chainedTasks = new HashMap<>(serializedTasks.size());
     serializedTasks.forEach(
         (id, config) -> chainedTasks.put(id, new StreamConfig(config.getConfiguration())));
-    StreamConfig taskConfig = vertexConfig;
-    while (true) {
-      List<StreamEdge> outEdges = taskConfig.getChainedOutputs(userClassloader);
-      if (outEdges == null || outEdges.size() != 1) {
-        // only support operators have one output.
-        LOG.debug("{} has no or more than one chained task.", taskConfig.getOperatorName());
-        break;
-      }
+    buildGlutenChains(vertexConfig, chainedTasks);
+    // TODO: may need fallback if failed.
+    vertexConfig.setAndSerializeTransitiveChainedTaskConfigs(chainedTasks);
+  }
 
-      StreamEdge outEdge = outEdges.get(0);
+  private void buildGlutenChains(StreamConfig taskConfig, Map<Integer, StreamConfig> chainedTasks) {
+    boolean isGlutenOp = isGlutenOperator(taskConfig);
+    List<StreamEdge> outEdges = taskConfig.getChainedOutputs(userClassloader);
+    GlutenOperator sourceOperator = isGlutenOp ? getGlutenOperator(taskConfig) : null;
```

Comment:
Seems better to modify `getGlutenOperator` to return Optional<>? Then, the result can indicate it's gluten operator or not , and `isGlutenOperator` can be removed.